### PR TITLE
Story/vspc 196

### PR DESCRIPTION
**Pull request by pradeep-017**
# Guidelines for Pull Requests

If you haven't yet read our code review guidelines, please do so, You can find them [here](https://diging.atlassian.net/wiki/spaces/DIGING/pages/2256076801/Code+Review+Guidelines).

Please confirm the following by adding an x for each item (turn `[ ]` into `[x]`).

- [x] I have removed all code style changes that are not necessary (e.g. changing blanks across the whole file that don’t need to be changed, adding empty lines in parts other than your own code)
- [x] I am not making any changes to files that don’t have any effect (e.g. imports added that don’t need to be added)
- [x] I do not have any sysout statements in my code or commented out code that isn’t needed anymore
- [x] I am not reformatting any files in the wrong format or without cause. 
- [x] I am not changing file encoding or line endings to something else than UTF-8, LF
- [x] My pull request does not show an insane amount of files being changed although my ticket only requires a few files being changed
- [x] I have added Javadoc/documentation where appropriate
- [x] I have added test cases where appropriate
- [x] I have explained any part of my code/implementation decisions that is not be self-explanatory

## Please provide a brief description of your ticket 
(you can copy the ticket if it hasn't changed) 

Using existing images for space links, module links and external links
  
## Anything else the reviewer needs to know?

... describe here ...

### CODE REVIEW

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/LinkManager.java [hunk 1]
```diff
@@ -100,15 +104,18 @@ protected void validateSpace(String id) throws SpaceDoesNotExistException{
         }
     }
 
-    protected void setDisplayProperties(ILinkDisplay linkDisplay,String id, float positionX,float positionY,int rotation, DisplayType displayType, byte[] linkImage, String imageFilename) throws ImageCouldNotBeStoredException {
+    protected void setDisplayProperties(ILinkDisplay linkDisplay,String id, float positionX,float positionY,int rotation, DisplayType displayType, byte[] linkImage, String imageFilename, String existingImageId) throws ImageCouldNotBeStoredException, ImageDoesNotExistException {
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/LinkManager.java:107]: this method should be overloaded (one method takes an image and one takes an image id).

File: vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html [hunk 2]
```diff
@@ -85,6 +85,7 @@
             }
                 
             function uploadImage() {
+            	debugger;
```
Comment (jdamerow) [inline vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html:88]: ?

**Review by jdamerow — CHANGES_REQUESTED** [review]
This also applies to javascript:
"I do not have any sysout statements in my code or commented out code that isn’t needed anymore"

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/LinkManager.java [hunk 3]
```diff
@@ -119,8 +134,20 @@ protected void setDisplayProperties(ILinkDisplay linkDisplay,String id, float po
             image.setParentPath(relativePath);
             imageRepo.save((VSImage) image);
             linkDisplay.setImage(image);
+        } 
+        
+    }
+    
+    protected void setDisplayProperties(ILinkDisplay linkDisplay,String id, float positionX,float positionY,int rotation, DisplayType displayType, byte[] linkImage, String imageFilename, String existingImageId) throws ImageCouldNotBeStoredException, ImageDoesNotExistException {
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/LinkManager.java:141]: there are a bunch of parameters here that don't seem to be used in this method.

File: vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SpaceLinkManagerTest.java [hunk 4]
```diff
@@ -218,12 +281,43 @@ public void test_updateLink_success() throws SpaceDoesNotExistException, LinkDoe
...
+        ISpaceLinkDisplay actualUpdatedLink = managerToTest.updateLink("TestSpaceEdited", spaceId1, 100, 300, 180, spaceId3, "TestSpaceEdited", "SPL002", "SPLD001", DisplayType.ALERT, null, null, null);
         Assert.assertEquals(spaceLinkDisplayUpdated.getId(), actualUpdatedLink.getId());
         Assert.assertEquals(spaceLinkDisplayUpdated.getName(), actualUpdatedLink.getName());
         Assert.assertEquals(new Double(spaceLinkDisplayUpdated.getPositionX()), new Double(actualUpdatedLink.getPositionX()));
         Assert.assertEquals(new Double(spaceLinkDisplayUpdated.getPositionY()), new Double(actualUpdatedLink.getPositionY()));
         Assert.assertEquals(spaceLinkDisplayUpdated.getLink().getTargetSpace(), actualUpdatedLink.getLink().getTargetSpace());
         Assert.assertEquals(spaceLinkDisplayUpdated.getType(), actualUpdatedLink.getType());
+        
+        
+        //For spaceLink updating with existing Image
```
Comment (jdamerow) [inline vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SpaceLinkManagerTest.java:293]: that should be its own test

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by pradeep-017** [comment]
Closing this. Will be tracked in https://github.com/diging/virtual-spaces-2.0/pull/247

**Pull request closed without merging**